### PR TITLE
refactor: acl express fastify

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-orchestra/issues/16 |
| Dependencies | -- |
| Decisions | Refactor common logic out of fastify module. Preferably we would have fastifyAcl and expressAcl but that is a breaking change for clients using just "acl". Also create the express module and handlers. |
| Animated GIF | -- |

### CODE REVIEW

**Review by 3rdvision — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,11 +9,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-*
+* Express middleware not found and error handlers - [ripe-orchestra/#16](https://github.com/ripe-tech/ripe-orchestra/issues/16)
 
 ### Changed
 
-*
+* Refactored the ACL function out of the fastify module to be used by express aswell - [ripe-orchestra/#16](https://github.com/ripe-tech/ripe-orchestra/issues/16)
```
Comment (BeeMargarida) [inline CHANGELOG.md:16]:
```suggestion
* Refactored the ACL function out of the fastify module to be used by express as well - [ripe-orchestra/#16](https://github.com/ripe-tech/ripe-orchestra/issues/16)
```

**Review by BeeMargarida — COMMENTED** [review]
just a small thing in the docs

**Review by BeeMargarida — APPROVED** [review]

**Comment by joamag** [comment]
We need to refactor this solution as we'll face type challenges with having an abstract req, res that handles both Express and Fastify. Let's not forget that Typescript does not handle duck typing :)
@joao-conde let's talk 5 min about this one

**Review by joamag — APPROVED** [review]

**Comment by joao-conde** [comment]
True, forgot about TS. The ideal would be having acl functions for both fastify and express but we have colliding exports if both are exported as acl. And if we change to aclFastify and aclExpres we will have breaking changes.

**Comment by joamag** [comment]
> True, forgot about TS. The ideal would be having acl functions for both fastify and express but we have colliding exports if both are exported as acl. And if we change to aclFastify and aclExpres we will have breaking changes.

yes, that's correct but because we'll change the name of those functions and they are exported we'll need to make sure that we have no regressions in projects that depend on the `acl()` function.

**Pull request merged**